### PR TITLE
Make "Write functions, not just scripts" more prominent in the performance tips

### PR DESCRIPTION
**Pull request by mschauer**
Make "Write functions, not just scripts" more prominent in the performance tips. This came up on Slack. @dpsanders

### CODE REVIEW

File: doc/src/manual/performance-tips.md [hunk 1]
```diff
@@ -3,13 +3,20 @@
 In the following sections, we briefly go through a few techniques that can help make your Julia
 code run as fast as possible.
 
+## Write functions, not just scripts
+
+Any code that is performance critical should be inside a function. Code inside functions tends to run much faster than top level code, due to how Julia's compiler works.
+
+The use of functions is not only important for performance: functions are more reusable and testable, and clarify what steps are being done and what their inputs and outputs are, [Write functions, not just scripts](@ref) is also a recommendation of Julia's Styleguide.
```
Comment (mschauer) [inline doc/src/manual/performance-tips.md:10]: Unsure about the right syntax here.

File: doc/src/manual/performance-tips.md [hunk 2]
```diff
@@ -3,13 +3,20 @@
 In the following sections, we briefly go through a few techniques that can help make your Julia
 code run as fast as possible.
 
+## Write functions, not just scripts
```
Comment (mschauer) [inline doc/src/manual/performance-tips.md:6]:
```suggestion
## Performance critical code should be inside a function
```

Comment (mschauer) [reply doc/src/manual/performance-tips.md:6]: (otherwise the headers aren't unique)

**Comment by mschauer** [comment]
Doctest Passes.

**Review by carstenbauer — APPROVED** [review]

**Review by DilumAluthge — APPROVED** [review]

**Comment by DilumAluthge** [comment]
This looks good to me, and it doesn't seem like there are any objections to merging this.

I'll merge this now. If there are serious objections, we can revert it.

**Pull request merged**